app/javascript/components/Analytics/SelectPopover.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from "react";
+
+import { Icon } from "$app/components/Icons";
+import { Popover } from "$app/components/Popover";
+
+interface SelectOption<T extends string> {
+  value: T;
+  label: string;
+}
+
+interface SelectPopoverProps<T extends string> {
+  options: SelectOption<T>[];
+  value: T;
+  onChange: (value: T) => void;
+  ariaLabel?: string;
+  size?: "small" | "large";
+}
+
+export function SelectPopover<T extends string>({ options, value, onChange, ariaLabel, size = "small" }: SelectPopoverProps<T>) {
+  const selectedOption = options.find((opt) => opt.value === value);
+  const radioGroupName = React.useId();
+
+  const handleInputChange = (close: () => void) => (event: React.ChangeEvent<HTMLInputElement>) => {
+    const target = event.target as HTMLInputElement;
+    const optionValue = target.value as T;
+    onChange(optionValue);
+    close();
+  };
+
+  const widthClass = size === "small" ? "!w-32" : "!w-40";
```

Comment:
This feels arbitrary. It would be much simpler and more intuitive for this component to accept a `triggerClassName` prop so we can pass as many CSS classes as we want to the trigger when calling the component.